scripts/2022_10_31_opt_update_minichef_for_week_11_02__11_09.py

```diff
@@ -0,0 +1,98 @@
+import math
+
+from brownie import ZERO_ADDRESS, Contract, accounts, history, network
+from helpers import (CHAIN_IDS, MINICHEF_ADDRESSES,
+                     SIDECHAIN_TOTAL_EMISSION_RATE)
+
+TARGET_NETWORK = "OPTIMISM"
+
+
+def main():
+    """
+    Updates minichef parameters on Optimism network for week 11/04 - 11/11
+    Vote: https://snapshot.org/#/saddlefinance.eth/proposal/0x737703080fc3affa33fdf6e7205fff0ce5919dbf50f02e864138d1f70d105ce3
+    """
+    print(f"You are using the '{network.show_active()}' network")
+    assert (
+        network.chain.id == CHAIN_IDS[TARGET_NETWORK]
+    ), f"Not on {TARGET_NETWORK} network"
+
+    deployer = accounts.load("deployer")  # prompts for password
+
+    ##### Update Minichef weights #####
+    minichef = Contract.from_explorer(
+        MINICHEF_ADDRESSES[CHAIN_IDS[TARGET_NETWORK]], None, deployer
+    )
+
+    # Prepare an array of calldata to execute using batch function
+    batch_calldata = []
+
+    # Mass update pools to checkpoint existing stakers
+    pids = [1, 2, 3, 4, 5, 7]
+    batch_calldata.append(minichef.massUpdatePools.encode_input(pids))
+
+    # 35.90% for Optimism: FraxBP
+    assert minichef.lpToken(1) == "0xf74ebe6e5586275dc4CeD78F5DBEF31B1EfbE7a5"
+    if (minichef.poolInfo(1)[2] != 3590):
+        batch_calldata.append(minichef.set.encode_input(
+            1, 3590, ZERO_ADDRESS, False))
+
+    # 4.41% for Optimism: frax-optUSD
+    assert minichef.lpToken(2) == "0xfF5fa61Eb9b5cDD63bdFa16EF029d5313457925A"
+    if (minichef.poolInfo(2)[2] != 441):
+        batch_calldata.append(minichef.set.encode_input(
+            2, 441, ZERO_ADDRESS, False))
+
+    # 2.65% for Optimism: Frax-USDT
+    assert minichef.lpToken(3) == "0xb63d7B0D835ca6eFf89ab774498ed6dD0D71e93e"
+    if (minichef.poolInfo(3)[2] != 265):
+        batch_calldata.append(minichef.set.encode_input(
+            3, 265, ZERO_ADDRESS, False))
+
+    # 4.46% for Optimism: Frax-SUSD
```

Comment:
```suggestion
    # 4.41% for Optimism: Frax-SUSD
```
NIT: corrected comment